### PR TITLE
Add check for insecure hash algorithms

### PR DESCRIPTION
**Pull request by spraints**
MD5 and SHA1 are both subject to well-known attacks and are not recommended for use. This branch adds a cop that checks for these hash functions and recommends SHA256 in their place.

cc @dbussink @jhawthorn

### CODE REVIEW

**Comment by dbussink** [comment]
@spraints In https://github.com/github/github/pull/165161#discussion_r540253611 @vcsjones had a good idea to use an allow list instead. How about also doing that for the generic Rubocop rules here as well?

File: lib/rubocop/cop/github/insecure_hash_algorithm.rb [hunk 1]
```diff
@@ -6,33 +6,89 @@ module RuboCop
...
+        def just_encoding?(val)
+          val == :hexencode || val == :bubblebabble
+        end
+
+        # Built-in hash functions are listed in these docs:
+        #  https://ruby-doc.org/stdlib-2.7.0/libdoc/digest/rdoc/Digest.html
+        #  https://ruby-doc.org/stdlib-2.7.0/libdoc/openssl/rdoc/OpenSSL/Digest.html
+        DEFAULT_ALLOWED = %w[
+          RIPEMD160
+          RMD160
```
Comment (dbussink) [inline lib/rubocop/cop/github/insecure_hash_algorithm.rb:63]: These shouldn't be allowed.

Comment (spraints) [reply lib/rubocop/cop/github/insecure_hash_algorithm.rb]: I removed them (and SHA-224). The default allow list now is:

https://github.com/github/rubocop-github/blob/4d952bbb1da770bfe49586a4db2a79061eed612a/lib/rubocop/cop/github/insecure_hash_algorithm.rb#L61-L65

**Review by dbussink — APPROVED** [review]
Looking good to me :+1:

**Review by jhawthorn — APPROVED** [review]

**Pull request merged**